modules/edu.gemini.seqexec.engine/src/test/scala/edu/gemini/seqexec/engine/StepSpec.scala

```diff
@@ -185,18 +195,189 @@ class StepSpec extends FlatSpec {
 
   }
 
-  ignore should "ignore pause command if step is not been executed." in {
+  it should "cancel a pause request in response to a cancel pause command." in {
+    val qs0: Engine.State =
+      Engine.State(
+        Conditions.default,
+        None,
+        Map(
+          (seqId,
+           Sequence.State.Zipper(
+             Sequence.Zipper(
+               "First",
+               SequenceMetadata(F2, None, ""),
+               Nil,
+               Step.Zipper(
+                 2,
+                 None,
+                 config,
+                 Set.empty,
+                 false,
+                 false,
+                 Nil,
+                 Execution(List(observe.left)),
+                 List(List(result, result)),
+                 (Execution(List(configureTcs.left, configureInst.left)), List(List(observe)))),
+               Nil
+             ),
+             SequenceState.Stopping
+           )
+          )
+        )
+      )
+
+    val qs1 = process(Process.eval(Task.now(Event.cancelPause(seqId, user))))(qs0).take(1).runLast.unsafePerformSync.map(_._2)
+
+    inside (qs1.flatMap(_.sequences.get(seqId))) {
+      case Some(Sequence.State.Zipper(zipper, status)) =>
+        status should be (Running)
+    }
+
+  }
+
+  "engine" should "ignore pause command if step is not being executed." in {
+    val qs0: Engine.State =
+      Engine.State(
+        Conditions.default,
+        None,
+        Map(
+          (seqId,
+           Sequence.State.init(
+             Sequence(
+               seqId,
+               SequenceMetadata(F2, None, ""),
+               List(
+                 Step(
+                   1,
+                   None,
+                   config,
+                   Set.empty,
+                   false,
+                   false,
+                   List(
+                     List(configureTcs, configureInst).map(_.left), // Execution
+                     List(observe.left) // Execution
+                   )
+                 )
+               )
+             )
+           )
+          )
+        )
+      )
+
+    val qss = process(Process.eval(Task.now(Event.pause(seqId, user))))(qs0).runLog.unsafePerformSync.map(_._2)
 
+    assert(qss.length == 1)
+    inside (qss.headOption.flatMap(_.sequences.get(seqId))) {
+      case Some(Sequence.State.Zipper(zipper, status)) =>
+        inside (zipper.focus.toStep) {
+          case Step(_, _, _, _, _, _, ex1::ex2::Nil) =>
```

Comment:
i suppose there are other possible cases, Will the fail the test?. is that assured by `inside`?